packages/typescript/types/index.d.ts

```diff
@@ -38,40 +70,21 @@ export interface RollupTypescriptPluginOptions {
   transformers?: CustomTransformerFactories;
 }
 
-type ElementType<T extends Array<any> | undefined> = T extends (infer U)[] ? U : never;
-
-export type TransformerStage = keyof CustomTransformers;
-type StagedTransformerFactory<T extends TransformerStage> = ElementType<CustomTransformers[T]>;
-type TransformerFactory<T extends TransformerStage> =
-  | StagedTransformerFactory<T>
-  | ProgramTransformerFactory<T>
-  | TypeCheckerTransformerFactory<T>;
-
-export type CustomTransformerFactories = {
-  [stage in TransformerStage]?: Array<TransformerFactory<stage>>;
-};
-
-interface ProgramTransformerFactory<T extends TransformerStage> {
-  type: 'program';
-
-  factory(program: Program): StagedTransformerFactory<T>;
-}
-
-interface TypeCheckerTransformerFactory<T extends TransformerStage> {
-  type: 'typeChecker';
-
-  factory(typeChecker: TypeChecker): StagedTransformerFactory<T>;
+export interface FlexibleCompilerOptions extends CompilerOptions {
+  [option: string]: CompilerOptionsValue | TsConfigSourceFile | undefined | any;
 }
 
 /** Properties of `CompilerOptions` that are normally enums */
 export type EnumCompilerOptions = 'module' | 'moduleResolution' | 'newLine' | 'jsx' | 'target';
 
 /** JSON representation of Typescript compiler options */
-export type JsonCompilerOptions = Omit<CompilerOptions, EnumCompilerOptions> &
+export type JsonCompilerOptions = Omit<FlexibleCompilerOptions, EnumCompilerOptions> &
```

Comment:
The type changes here appear unnecessary, and throws out a lot of type-checking for typos. I'd rather leave it as is.

Reply:
Unfortunately it necessary for everything to play nicely together. The tests weren't written in TypeScript, and as soon as  I started passing `tslib` as an option in a TypeScript test, it started throwing typing errors. Going to have to overrule you on this one, unless you can get the types playing nice in a more visually pleasing manner.

Reply:
OK, I can always try taking a look at this again in a future PR

Reply:
Sorry for laying that out tersely, super tired over here. The blame is not yours, so didn't mean to imply that. Just how the types ended up, we got that funky collision. We'd probably have to use an alternate option name like `tslibPath` or something.